test/convenience-constructor-test.js

```diff
@@ -0,0 +1,355 @@
+					flushed = true;
+					cb();
+				}
+			});
+			zstreams.fromString('abcd')
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(flushed).to.be.true;
+					done();
+				});
+		});
+	});
+	describe('object mode', function() {
+		it('should verify flush is called', function(done) {
+			var writeStream = new Writable({
+				objectMode: true,
+				write: function(chunk, encoding, cb) {
+					cb();
+				},
+				flush: function(cb) {
+					flushed = true;
+					cb();
+				}
+			});
+			zstreams.fromString('abcd')
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(flushed).to.be.true;
+					done();
+				});
+		});
+	});
+});
+
+describe('Duplex._read', function() {
+	describe('data mode', function() {
+		it('should match what the writable contains', function(done) {
+			var duplexStream = new Duplex({
+				read: function() {
+					this.push('a');
+					this.push('b');
+					this.push('c');
+					this.push('d');
+					this.push(null);
+				}
+			});
+			duplexStream.intoString(function(error, str) {
+				expect(error).to.not.exist;
+				expect(str).to.have.length(4);
+				expect(str).to.be.a('string');
+				done();
+			});
+		});
+	});
+
+	describe('object mode', function() {
+		it('should match what the writable contains', function(done) {
+			var duplexStream = new Duplex({
+				objectMode: true,
+				read: function() {
+					this.push('a');
+					this.push('b');
+					this.push('c');
+					this.push('d');
+					this.push(null);
+				}
+			});
+			duplexStream.intoString(function(error, str) {
+				expect(error).to.not.exist;
+				expect(str).to.have.length(4);
+				expect(str).to.be.a('string');
+				done();
+			});
+		});
+	});
+});
+
+
+describe('Duplex._write', function () {
+	describe('data mode', function() {
+		it('should receive the Readable stream', function(done) {
+			var duplexStream = new Duplex({
+				write: function(chunk, encoding, cb) {
+					count++;
+					cb();
+				}
+			});
+			var count = 0;
+			zstreams.fromString('abcd', {chunkSize: 1})
+				.pipe(duplexStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(count).to.equal(4);
+					done();
+				});
+		});
+	});
+	describe('object mode', function() {
+		it('should receive the Readable stream', function(done) {
+			var duplexStream = new Duplex({
+				objectMode: true,
+				write: function(chunk, encoding, cb) {
+					count++
+					cb();
+				}
+			});
+			var count = 0;
+			zstreams.fromString('abcd', {chunkSize: 1})
+				.pipe(duplexStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(count).to.equal(4);
+					done();
+				});
+		});
+	});
+});
+
+describe('Duplex._flush', function () {
+	describe('data mode', function() {
+		it('should verify flush is called', function(done) {
+			var flushed = false;
+			var duplexStream = new Duplex({
+				write: function(chunk, encoding, cb) {
```

Comment:
Or rather, combine the read/write trests. A duplex should implement both, especially if piped through a readable/writable.